// nuxi/dist/logger-CyBffPrB.mjs
import { a as __toCommonJS, i as __require, n as __esmMin, o as __toESM, r as __exportAll, t as __commonJSMin } from "./chunk-Vs_PY4HZ.mjs";
import "node:path";
import process$1, { stdin, stdout } from "node:process";
import { stripVTControlCharacters, styleText } from "node:util";
import tty, { ReadStream } from "node:tty";
import * as _ from "node:readline";
import u from "node:readline";
import "node:fs";
import os from "node:os";
//#region ../../node_modules/.pnpm/fast-string-truncated-width@1.2.1/node_modules/fast-string-truncated-width/dist/utils.js
const isAmbiguous = (x) => {
	return x === 161 || x === 164 || x === 167 || x === 168 || x === 170 || x === 173 || x === 174 || x >= 176 && x <= 180 || x >= 182 && x <= 186 || x >= 188 && x <= 191 || x === 198 || x === 208 || x === 215 || x === 216 || x >= 222 && x <= 225 || x === 230 || x >= 232 && x <= 234 || x === 236 || x === 237 || x === 240 || x === 242 || x === 243 || x >= 247 && x <= 250 || x === 252 || x === 254 || x === 257 || x === 273 || x === 275 || x === 283 || x === 294 || x === 295 || x === 299 || x >= 305 && x <= 307 || x === 312 || x >= 319 && x <= 322 || x === 324 || x >= 328 && x <= 331 || x === 333 || x === 338 || x === 339 || x === 358 || x === 359 || x === 363 || x === 462 || x === 464 || x === 466 || x === 468 || x === 470 || x === 472 || x === 474 || x === 476 || x === 593 || x === 609 || x === 708 || x === 711 || x >= 713 && x <= 715 || x === 717 || x === 720 || x >= 728 && x <= 731 || x === 733 || x === 735 || x >= 768 && x <= 879 || x >= 913 && x <= 929 || x >= 931 && x <= 937 || x >= 945 && x <= 961 || x >= 963 && x <= 969 || x === 1025 || x >= 1040 && x <= 1103 || x === 1105 || x === 8208 || x >= 8211 && x <= 8214 || x === 8216 || x === 8217 || x === 8220 || x === 8221 || x >= 8224 && x <= 8226 || x >= 8228 && x <= 8231 || x === 8240 || x === 8242 || x === 8243 || x === 8245 || x === 8251 || x === 8254 || x === 8308 || x === 8319 || x >= 8321 && x <= 8324 || x === 8364 || x === 8451 || x === 8453 || x === 8457 || x === 8467 || x === 8470 || x === 8481 || x === 8482 || x === 8486 || x === 8491 || x === 8531 || x === 8532 || x >= 8539 && x <= 8542 || x >= 8544 && x <= 8555 || x >= 8560 && x <= 8569 || x === 8585 || x >= 8592 && x <= 8601 || x === 8632 || x === 8633 || x === 8658 || x === 8660 || x === 8679 || x === 8704 || x === 8706 || x === 8707 || x === 8711 || x === 8712 || x === 8715 || x === 8719 || x === 8721 || x === 8725 || x === 8730 || x >= 8733 && x <= 8736 || x === 8739 || x === 8741 || x >= 8743 && x <= 8748 || x === 8750 || x >= 8756 && x <= 8759 || x === 8764 || x === 8765 || x === 8776 || x === 8780 || x === 8786 || x === 8800 || x === 8801 || x >= 8804 && x <= 8807 || x === 8810 || x === 8811 || x === 8814 || x === 8815 || x === 8834 || x === 8835 || x === 8838 || x === 8839 || x === 8853 || x === 8857 || x === 8869 || x === 8895 || x === 8978 || x >= 9312 && x <= 9449 || x >= 9451 && x <= 9547 || x >= 9552 && x <= 9587 || x >= 9600 && x <= 9615 || x >= 9618 && x <= 9621 || x === 9632 || x === 9633 || x >= 9635 && x <= 9641 || x === 9650 || x === 9651 || x === 9654 || x === 9655 || x === 9660 || x === 9661 || x === 9664 || x === 9665 || x >= 9670 && x <= 9672 || x === 9675 || x >= 9678 && x <= 9681 || x >= 9698 && x <= 9701 || x === 9711 || x === 9733 || x === 9734 || x === 9737 || x === 9742 || x === 9743 || x === 9756 || x === 9758 || x === 9792 || x === 9794 || x === 9824 || x === 9825 || x >= 9827 && x <= 9829 || x >= 9831 && x <= 9834 || x === 9836 || x === 9837 || x === 9839 || x === 9886 || x === 9887 || x === 9919 || x >= 9926 && x <= 9933 || x >= 9935 && x <= 9939 || x >= 9941 && x <= 9953 || x === 9955 || x === 9960 || x === 9961 || x >= 9963 && x <= 9969 || x === 9972 || x >= 9974 && x <= 9977 || x === 9979 || x === 9980 || x === 9982 || x === 9983 || x === 10045 || x >= 10102 && x <= 10111 || x >= 11094 && x <= 11097 || x >= 12872 && x <= 12879 || x >= 57344 && x <= 63743 || x >= 65024 && x <= 65039 || x === 65533 || x >= 127232 && x <= 127242 || x >= 127248 && x <= 127277 || x >= 127280 && x <= 127337 || x >= 127344 && x <= 127373 || x === 127375 || x === 127376 || x >= 127387 && x <= 127404 || x >= 917760 && x <= 917999 || x >= 983040 && x <= 1048573 || x >= 1048576 && x <= 1114109;
};
const isFullWidth = (x) => {
	return x === 12288 || x >= 65281 && x <= 65376 || x >= 65504 && x <= 65510;
};
const isWide = (x) => {
	return x >= 4352 && x <= 4447 || x === 8986 || x === 8987 || x === 9001 || x === 9002 || x >= 9193 && x <= 9196 || x === 9200 || x === 9203 || x === 9725 || x === 9726 || x === 9748 || x === 9749 || x >= 9800 && x <= 9811 || x === 9855 || x === 9875 || x === 9889 || x === 9898 || x === 9899 || x === 9917 || x === 9918 || x === 9924 || x === 9925 || x === 9934 || x === 9940 || x === 9962 || x === 9970 || x === 9971 || x === 9973 || x === 9978 || x === 9981 || x === 9989 || x === 9994 || x === 9995 || x === 10024 || x === 10060 || x === 10062 || x >= 10067 && x <= 10069 || x === 10071 || x >= 10133 && x <= 10135 || x === 10160 || x === 10175 || x === 11035 || x === 11036 || x === 11088 || x === 11093 || x >= 11904 && x <= 11929 || x >= 11931 && x <= 12019 || x >= 12032 && x <= 12245 || x >= 12272 && x <= 12287 || x >= 12289 && x <= 12350 || x >= 12353 && x <= 12438 || x >= 12441 && x <= 12543 || x >= 12549 && x <= 12591 || x >= 12593 && x <= 12686 || x >= 12688 && x <= 12771 || x >= 12783 && x <= 12830 || x >= 12832 && x <= 12871 || x >= 12880 && x <= 19903 || x >= 19968 && x <= 42124 || x >= 42128 && x <= 42182 || x >= 43360 && x <= 43388 || x >= 44032 && x <= 55203 || x >= 63744 && x <= 64255 || x >= 65040 && x <= 65049 || x >= 65072 && x <= 65106 || x >= 65108 && x <= 65126 || x >= 65128 && x <= 65131 || x >= 94176 && x <= 94180 || x === 94192 || x === 94193 || x >= 94208 && x <= 100343 || x >= 100352 && x <= 101589 || x >= 101632 && x <= 101640 || x >= 110576 && x <= 110579 || x >= 110581 && x <= 110587 || x === 110589 || x === 110590 || x >= 110592 && x <= 110882 || x === 110898 || x >= 110928 && x <= 110930 || x === 110933 || x >= 110948 && x <= 110951 || x >= 110960 && x <= 111355 || x === 126980 || x === 127183 || x === 127374 || x >= 127377 && x <= 127386 || x >= 127488 && x <= 127490 || x >= 127504 && x <= 127547 || x >= 127552 && x <= 127560 || x === 127568 || x === 127569 || x >= 127584 && x <= 127589 || x >= 127744 && x <= 127776 || x >= 127789 && x <= 127797 || x >= 127799 && x <= 127868 || x >= 127870 && x <= 127891 || x >= 127904 && x <= 127946 || x >= 127951 && x <= 127955 || x >= 127968 && x <= 127984 || x === 127988 || x >= 127992 && x <= 128062 || x === 128064 || x >= 128066 && x <= 128252 || x >= 128255 && x <= 128317 || x >= 128331 && x <= 128334 || x >= 128336 && x <= 128359 || x === 128378 || x === 128405 || x === 128406 || x === 128420 || x >= 128507 && x <= 128591 || x >= 128640 && x <= 128709 || x === 128716 || x >= 128720 && x <= 128722 || x >= 128725 && x <= 128727 || x >= 128732 && x <= 128735 || x === 128747 || x === 128748 || x >= 128756 && x <= 128764 || x >= 128992 && x <= 129003 || x === 129008 || x >= 129292 && x <= 129338 || x >= 129340 && x <= 129349 || x >= 129351 && x <= 129535 || x >= 129648 && x <= 129660 || x >= 129664 && x <= 129672 || x >= 129680 && x <= 129725 || x >= 129727 && x <= 129733 || x >= 129742 && x <= 129755 || x >= 129760 && x <= 129768 || x >= 129776 && x <= 129784 || x >= 131072 && x <= 196605 || x >= 196608 && x <= 262141;
};
//#endregion
//#region ../../node_modules/.pnpm/fast-string-truncated-width@1.2.1/node_modules/fast-string-truncated-width/dist/index.js
const ANSI_RE = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/y;
const CONTROL_RE = /[\x00-\x08\x0A-\x1F\x7F-\x9F]{1,1000}/y;
const TAB_RE = /\t{1,1000}/y;
const EMOJI_RE = /[\u{1F1E6}-\u{1F1FF}]{2}|\u{1F3F4}[\u{E0061}-\u{E007A}]{2}[\u{E0030}-\u{E0039}\u{E0061}-\u{E007A}]{1,3}\u{E007F}|(?:\p{Emoji}\uFE0F\u20E3?|\p{Emoji_Modifier_Base}\p{Emoji_Modifier}?|\p{Emoji_Presentation})(?:\u200D(?:\p{Emoji_Modifier_Base}\p{Emoji_Modifier}?|\p{Emoji_Presentation}|\p{Emoji}\uFE0F\u20E3?))*/uy;
const LATIN_RE = /(?:[\x20-\x7E\xA0-\xFF](?!\uFE0F)){1,1000}/y;
const MODIFIER_RE = /\p{M}+/gu;
const NO_TRUNCATION$1 = {
	limit: Infinity,
	ellipsis: ""
};
const getStringTruncatedWidth = (input, truncationOptions = {}, widthOptions = {}) => {
	const LIMIT = truncationOptions.limit ?? Infinity;
	const ELLIPSIS = truncationOptions.ellipsis ?? "";
	const ELLIPSIS_WIDTH = truncationOptions?.ellipsisWidth ?? (ELLIPSIS ? getStringTruncatedWidth(ELLIPSIS, NO_TRUNCATION$1, widthOptions).width : 0);
	const ANSI_WIDTH = widthOptions.ansiWidth ?? 0;
	const CONTROL_WIDTH = widthOptions.controlWidth ?? 0;
	const TAB_WIDTH = widthOptions.tabWidth ?? 8;
	const AMBIGUOUS_WIDTH = widthOptions.ambiguousWidth ?? 1;
	const EMOJI_WIDTH = widthOptions.emojiWidth ?? 2;
	const FULL_WIDTH_WIDTH = widthOptions.fullWidthWidth ?? 2;
	const REGULAR_WIDTH = widthOptions.regularWidth ?? 1;
	const WIDE_WIDTH = widthOptions.wideWidth ?? 2;
	let indexPrev = 0;
	let index = 0;
	let length = input.length;
	let lengthExtra = 0;
	let truncationEnabled = false;
	let truncationIndex = length;
	let truncationLimit = Math.max(0, LIMIT - ELLIPSIS_WIDTH);
	let unmatchedStart = 0;
	let unmatchedEnd = 0;
	let width = 0;
	let widthExtra = 0;
	outer: while (true) {
		if (unmatchedEnd > unmatchedStart || index >= length && index > indexPrev) {
			const unmatched = input.slice(unmatchedStart, unmatchedEnd) || input.slice(indexPrev, index);
			lengthExtra = 0;
			for (const char of unmatched.replaceAll(MODIFIER_RE, "")) {
				const codePoint = char.codePointAt(0) || 0;
				if (isFullWidth(codePoint)) widthExtra = FULL_WIDTH_WIDTH;
				else if (isWide(codePoint)) widthExtra = WIDE_WIDTH;
				else if (AMBIGUOUS_WIDTH !== REGULAR_WIDTH && isAmbiguous(codePoint)) widthExtra = AMBIGUOUS_WIDTH;
				else widthExtra = REGULAR_WIDTH;
				if (width + widthExtra > truncationLimit) truncationIndex = Math.min(truncationIndex, Math.max(unmatchedStart, indexPrev) + lengthExtra);
				if (width + widthExtra > LIMIT) {
					truncationEnabled = true;
					break outer;
				}
				lengthExtra += char.length;
				width += widthExtra;
			}
			unmatchedStart = unmatchedEnd = 0;
		}
		if (index >= length) break;
		LATIN_RE.lastIndex = index;
		if (LATIN_RE.test(input)) {
			lengthExtra = LATIN_RE.lastIndex - index;
			widthExtra = lengthExtra * REGULAR_WIDTH;
			if (width + widthExtra > truncationLimit) truncationIndex = Math.min(truncationIndex, index + Math.floor((truncationLimit - width) / REGULAR_WIDTH));
			if (width + widthExtra > LIMIT) {
				truncationEnabled = true;
				break;
			}
			width += widthExtra;
			unmatchedStart = indexPrev;
			unmatchedEnd = index;
			index = indexPrev = LATIN_RE.lastIndex;
			continue;
		}
		ANSI_RE.lastIndex = index;
		if (ANSI_RE.test(input)) {
			if (width + ANSI_WIDTH > truncationLimit) truncationIndex = Math.min(truncationIndex, index);
			if (width + ANSI_WIDTH > LIMIT) {
				truncationEnabled = true;
				break;
			}
			width += ANSI_WIDTH;
			unmatchedStart = indexPrev;
			unmatchedEnd = index;
			index = indexPrev = ANSI_RE.lastIndex;
			continue;
		}
		CONTROL_RE.lastIndex = index;
		if (CONTROL_RE.test(input)) {
			lengthExtra = CONTROL_RE.lastIndex - index;
			widthExtra = lengthExtra * CONTROL_WIDTH;
			if (width + widthExtra > truncationLimit) truncationIndex = Math.min(truncationIndex, index + Math.floor((truncationLimit - width) / CONTROL_WIDTH));
			if (width + widthExtra > LIMIT) {
				truncationEnabled = true;
				break;
			}
			width += widthExtra;
			unmatchedStart = indexPrev;
			unmatchedEnd = index;
			index = indexPrev = CONTROL_RE.lastIndex;
			continue;
		}
		TAB_RE.lastIndex = index;
		if (TAB_RE.test(input)) {
			lengthExtra = TAB_RE.lastIndex - index;
			widthExtra = lengthExtra * TAB_WIDTH;
			if (width + widthExtra > truncationLimit) truncationIndex = Math.min(truncationIndex, index + Math.floor((truncationLimit - width) / TAB_WIDTH));
			if (width + widthExtra > LIMIT) {
				truncationEnabled = true;
				break;
			}
			width += widthExtra;
			unmatchedStart = indexPrev;
			unmatchedEnd = index;
			index = indexPrev = TAB_RE.lastIndex;
			continue;
		}
		EMOJI_RE.lastIndex = index;
		if (EMOJI_RE.test(input)) {
			if (width + EMOJI_WIDTH > truncationLimit) truncationIndex = Math.min(truncationIndex, index);
			if (width + EMOJI_WIDTH > LIMIT) {
				truncationEnabled = true;
				break;
			}
			width += EMOJI_WIDTH;
			unmatchedStart = indexPrev;
			unmatchedEnd = index;
			index = indexPrev = EMOJI_RE.lastIndex;
			continue;
		}
		index += 1;
	}
	return {
		width: truncationEnabled ? truncationLimit : width,
		index: truncationEnabled ? truncationIndex : length,
		truncated: truncationEnabled,
		ellipsed: truncationEnabled && LIMIT >= ELLIPSIS_WIDTH
	};
};
//#endregion
//#region ../../node_modules/.pnpm/fast-string-width@1.1.0/node_modules/fast-string-width/dist/index.js
const NO_TRUNCATION = {
	limit: Infinity,
	ellipsis: "",
	ellipsisWidth: 0
};
const fastStringWidth = (input, options = {}) => {
	return getStringTruncatedWidth(input, NO_TRUNCATION, options).width;
};
//#endregion
//#region ../../node_modules/.pnpm/fast-wrap-ansi@0.1.6/node_modules/fast-wrap-ansi/lib/main.js
const ESC = "\x1B";
const CSI = "";
const END_CODE = 39;
const ANSI_ESCAPE_BELL = "\x07";
const ANSI_CSI = "[";
const ANSI_OSC = "]";
const ANSI_SGR_TERMINATOR = "m";
const ANSI_ESCAPE_LINK = `${ANSI_OSC}8;;`;
const GROUP_REGEX = new RegExp(`(?:\\${ANSI_CSI}(?<code>\\d+)m|\\${ANSI_ESCAPE_LINK}(?<uri>.*)${ANSI_ESCAPE_BELL})`, "y");
const getClosingCode = (openingCode) => {
	if (openingCode >= 30 && openingCode <= 37) return 39;
	if (openingCode >= 90 && openingCode <= 97) return 39;
	if (openingCode >= 40 && openingCode <= 47) return 49;
	if (openingCode >= 100 && openingCode <= 107) return 49;
	if (openingCode === 1 || openingCode === 2) return 22;
	if (openingCode === 3) return 23;
	if (openingCode === 4) return 24;
	if (openingCode === 7) return 27;
	if (openingCode === 8) return 28;
	if (openingCode === 9) return 29;
	if (openingCode === 0) return 0;
};
const wrapAnsiCode = (code) => `${ESC}${ANSI_CSI}${code}${ANSI_SGR_TERMINATOR}`;
const wrapAnsiHyperlink = (url) => `${ESC}${ANSI_ESCAPE_LINK}${url}${ANSI_ESCAPE_BELL}`;
const wrapWord = (rows, word, columns) => {
	const characters = word[Symbol.iterator]();
	let isInsideEscape = false;
	let isInsideLinkEscape = false;
	let lastRow = rows.at(-1);
	let visible = lastRow === void 0 ? 0 : fastStringWidth(lastRow);
	let currentCharacter = characters.next();
	let nextCharacter = characters.next();
	let rawCharacterIndex = 0;
	while (!currentCharacter.done) {
		const character = currentCharacter.value;
		const characterLength = fastStringWidth(character);
		if (visible + characterLength <= columns) rows[rows.length - 1] += character;
		else {
			rows.push(character);
			visible = 0;
		}
		if (character === ESC || character === CSI) {
			isInsideEscape = true;
			isInsideLinkEscape = word.startsWith(ANSI_ESCAPE_LINK, rawCharacterIndex + 1);
		}
		if (isInsideEscape) {
			if (isInsideLinkEscape) {
				if (character === ANSI_ESCAPE_BELL) {
					isInsideEscape = false;
					isInsideLinkEscape = false;
				}
			} else if (character === ANSI_SGR_TERMINATOR) isInsideEscape = false;
		} else {
			visible += characterLength;
			if (visible === columns && !nextCharacter.done) {
				rows.push("");
				visible = 0;
			}
		}
		currentCharacter = nextCharacter;
		nextCharacter = characters.next();
		rawCharacterIndex += character.length;
	}
	lastRow = rows.at(-1);
	if (!visible && lastRow !== void 0 && lastRow.length && rows.length > 1) rows[rows.length - 2] += rows.pop();
};
const stringVisibleTrimSpacesRight = (string) => {
	const words = string.split(" ");
	let last = words.length;
	while (last) {
		if (fastStringWidth(words[last - 1])) break;
		last--;
	}
	if (last === words.length) return string;
	return words.slice(0, last).join(" ") + words.slice(last).join("");
};
const exec = (string, columns, options = {}) => {
	if (options.trim !== false && string.trim() === "") return "";
	let returnValue = "";
	let escapeCode;
	let escapeUrl;
	const words = string.split(" ");
	let rows = [""];
	let rowLength = 0;
	for (let index = 0; index < words.length; index++) {
		const word = words[index];
		if (options.trim !== false) {
			const row = rows.at(-1) ?? "";
			const trimmed = row.trimStart();
			if (row.length !== trimmed.length) {
				rows[rows.length - 1] = trimmed;
				rowLength = fastStringWidth(trimmed);
			}
		}
		if (index !== 0) {
			if (rowLength >= columns && (options.wordWrap === false || options.trim === false)) {
				rows.push("");
				rowLength = 0;
			}
			if (rowLength || options.trim === false) {
				rows[rows.length - 1] += " ";
				rowLength++;
			}
		}
		const wordLength = fastStringWidth(word);
		if (options.hard && wordLength > columns) {
			const remainingColumns = columns - rowLength;
			const breaksStartingThisLine = 1 + Math.floor((wordLength - remainingColumns - 1) / columns);
			if (Math.floor((wordLength - 1) / columns) < breaksStartingThisLine) rows.push("");
			wrapWord(rows, word, columns);
			rowLength = fastStringWidth(rows.at(-1) ?? "");
			continue;
		}
		if (rowLength + wordLength > columns && rowLength && wordLength) {
			if (options.wordWrap === false && rowLength < columns) {
				wrapWord(rows, word, columns);
				rowLength = fastStringWidth(rows.at(-1) ?? "");
				continue;
			}
			rows.push("");
			rowLength = 0;
		}
		if (rowLength + wordLength > columns && options.wordWrap === false) {
			wrapWord(rows, word, columns);
			rowLength = fastStringWidth(rows.at(-1) ?? "");
			continue;
		}
		rows[rows.length - 1] += word;
		rowLength += wordLength;
	}
	if (options.trim !== false) rows = rows.map((row) => stringVisibleTrimSpacesRight(row));
	const preString = rows.join("\n");
	let inSurrogate = false;
	for (let i = 0; i < preString.length; i++) {
		const character = preString[i];
		returnValue += character;
		if (!inSurrogate) inSurrogate = character >= "\ud800" && character <= "\udbff";
		else continue;
		if (character === ESC || character === CSI) {
			GROUP_REGEX.lastIndex = i + 1;
			const groups = GROUP_REGEX.exec(preString)?.groups;
			if (groups?.code !== void 0) {
				const code = Number.parseFloat(groups.code);
				escapeCode = code === END_CODE ? void 0 : code;
			} else if (groups?.uri !== void 0) escapeUrl = groups.uri.length === 0 ? void 0 : groups.uri;
		}
		if (preString[i + 1] === "\n") {
			if (escapeUrl) returnValue += wrapAnsiHyperlink("");
			const closingCode = escapeCode ? getClosingCode(escapeCode) : void 0;
			if (escapeCode && closingCode) returnValue += wrapAnsiCode(closingCode);
		} else if (character === "\n") {
			if (escapeCode && getClosingCode(escapeCode)) returnValue += wrapAnsiCode(escapeCode);
			if (escapeUrl) returnValue += wrapAnsiHyperlink(escapeUrl);
		}
	}
	return returnValue;
};
const CRLF_OR_LF = /\r?\n/;
function wrapAnsi(string, columns, options) {
	return String(string).normalize().split(CRLF_OR_LF).map((line) => exec(line, columns, options)).join("\n");
}
//#endregion
//#region ../../node_modules/.pnpm/@clack+core@1.2.0/node_modules/@clack/core/dist/index.mjs
var import_src$1 = (/* @__PURE__ */ __commonJSMin(((exports, module) => {
	const ESC = "\x1B";
	const CSI = `${ESC}[`;
	const beep = "\x07";
	const cursor = {
		to(x, y) {
			if (!y) return `${CSI}${x + 1}G`;
			return `${CSI}${y + 1};${x + 1}H`;
		},
		move(x, y) {
			let ret = "";
			if (x < 0) ret += `${CSI}${-x}D`;
			else if (x > 0) ret += `${CSI}${x}C`;
			if (y < 0) ret += `${CSI}${-y}A`;
			else if (y > 0) ret += `${CSI}${y}B`;
			return ret;
		},
		up: (count = 1) => `${CSI}${count}A`,
		down: (count = 1) => `${CSI}${count}B`,
		forward: (count = 1) => `${CSI}${count}C`,
		backward: (count = 1) => `${CSI}${count}D`,
		nextLine: (count = 1) => `${CSI}E`.repeat(count),
		prevLine: (count = 1) => `${CSI}F`.repeat(count),
		left: `${CSI}G`,
		hide: `${CSI}?25l`,
		show: `${CSI}?25h`,
		save: `${ESC}7`,
		restore: `${ESC}8`
	};
	module.exports = {
		cursor,
		scroll: {
			up: (count = 1) => `${CSI}S`.repeat(count),
			down: (count = 1) => `${CSI}T`.repeat(count)
		},
		erase: {
			screen: `${CSI}2J`,
			up: (count = 1) => `${CSI}1J`.repeat(count),
			down: (count = 1) => `${CSI}J`.repeat(count),
			line: `${CSI}2K`,
			lineEnd: `${CSI}K`,
			lineStart: `${CSI}1K`,
			lines(count) {
				let clear = "";
				for (let i = 0; i < count; i++) clear += this.line + (i < count - 1 ? cursor.up() : "");
				if (count) clear += cursor.left;
				return clear;
			}
		},
		beep
	};
})))();
function d$1(r, t, e) {
	if (!e.some((o) => !o.disabled)) return r;
	const s = r + t, i = Math.max(e.length - 1, 0), n = s < 0 ? i : s > i ? 0 : s;
	return e[n].disabled ? d$1(n, t < 0 ? -1 : 1, e) : n;
}
const u$1 = {
	actions: new Set([
		"up",
		"down",
		"left",
		"right",
		"space",
		"enter",
		"cancel"
	]),
	aliases: new Map([
		["k", "up"],
		["j", "down"],
		["h", "left"],
		["l", "right"],
		["", "cancel"],
		["escape", "cancel"]
	]),
	messages: {
		cancel: "Canceled",
		error: "Something went wrong"
	},
	withGuide: !0,
	date: {
		monthNames: [...[
			"January",
			"February",
			"March",
			"April",
			"May",
			"June",
			"July",
			"August",
			"September",
			"October",
			"November",
			"December"
		]],
		messages: {
			required: "Please enter a valid date",
			invalidMonth: "There are only 12 months in a year",
			invalidDay: (r, t) => `There are only ${r} days in ${t}`,
			afterMin: (r) => `Date must be on or after ${r.toISOString().slice(0, 10)}`,
			beforeMax: (r) => `Date must be on or before ${r.toISOString().slice(0, 10)}`
		}
	}
};
function V$1(r, t) {
	if (typeof r == "string") return u$1.aliases.get(r) === t;
	for (const e of r) if (e !== void 0 && V$1(e, t)) return !0;
	return !1;
}
function j(r, t) {
	if (r === t) return;
	const e = r.split(`
`), s = t.split(`
`), i = Math.max(e.length, s.length), n = [];
	for (let o = 0; o < i; o++) e[o] !== s[o] && n.push(o);
	return {
		lines: n,
		numLinesBefore: e.length,
		numLinesAfter: s.length,
		numLines: i
	};
}
const Y$1 = globalThis.process.platform.startsWith("win"), C = Symbol("clack:cancel");
function q(r) {
	return r === C;
}
function w$1(r, t) {
	const e = r;
	e.isTTY && e.setRawMode(t);
}
function z$1({ input: r = stdin, output: t = stdout, overwrite: e = !0, hideCursor: s = !0 } = {}) {
	const i = _.createInterface({
		input: r,
		output: t,
		prompt: "",
		tabSize: 1
	});
	_.emitKeypressEvents(r, i), r instanceof ReadStream && r.isTTY && r.setRawMode(!0);
	const n = (o, { name: a, sequence: h }) => {
		if (V$1([
			String(o),
			a,
			h
		], "cancel")) {
			s && t.write(import_src$1.cursor.show), process.exit(0);
			return;
		}
		if (!e) return;
		const f = a === "return" ? 0 : -1, v = a === "return" ? -1 : 0;
		_.moveCursor(t, f, v, () => {
			_.clearLine(t, 1, () => {
				r.once("keypress", n);
			});
		});
	};
	return s && t.write(import_src$1.cursor.hide), r.once("keypress", n), () => {
		r.off("keypress", n), s && t.write(import_src$1.cursor.show), r instanceof ReadStream && r.isTTY && !Y$1 && r.setRawMode(!1), i.terminal = !1, i.close();
	};
}
const O$1 = (r) => "columns" in r && typeof r.columns == "number" ? r.columns : 80, A = (r) => "rows" in r && typeof r.rows == "number" ? r.rows : 20;
function R(r, t, e, s = e) {
	return wrapAnsi(t, O$1(r ?? stdout) - e.length, {
		hard: !0,
		trim: !1
	}).split(`
`).map((n, o) => `${o === 0 ? s : e}${n}`).join(`
`);
}
let p = class {
	input;
	output;
	_abortSignal;
	rl;
	opts;
	_render;
	_track = !1;
	_prevFrame = "";
	_subscribers = /* @__PURE__ */ new Map();
	_cursor = 0;
	state = "initial";
	error = "";
	value;
	userInput = "";
	constructor(t, e = !0) {
		const { input: s = stdin, output: i = stdout, render: n, signal: o, ...a } = t;
		this.opts = a, this.onKeypress = this.onKeypress.bind(this), this.close = this.close.bind(this), this.render = this.render.bind(this), this._render = n.bind(this), this._track = e, this._abortSignal = o, this.input = s, this.output = i;
	}
	unsubscribe() {
		this._subscribers.clear();
	}
	setSubscriber(t, e) {
		const s = this._subscribers.get(t) ?? [];
		s.push(e), this._subscribers.set(t, s);
	}
	on(t, e) {
		this.setSubscriber(t, { cb: e });
	}
	once(t, e) {
		this.setSubscriber(t, {
			cb: e,
			once: !0
		});
	}
	emit(t, ...e) {
		const s = this._subscribers.get(t) ?? [], i = [];
		for (const n of s) n.cb(...e), n.once && i.push(() => s.splice(s.indexOf(n), 1));
		for (const n of i) n();
	}
	prompt() {
		return new Promise((t) => {
			if (this._abortSignal) {
				if (this._abortSignal.aborted) return this.state = "cancel", this.close(), t(C);
				this._abortSignal.addEventListener("abort", () => {
					this.state = "cancel", this.close();
				}, { once: !0 });
			}
			this.rl = u.createInterface({
				input: this.input,
				tabSize: 2,
				prompt: "",
				escapeCodeTimeout: 50,
				terminal: !0
			}), this.rl.prompt(), this.opts.initialUserInput !== void 0 && this._setUserInput(this.opts.initialUserInput, !0), this.input.on("keypress", this.onKeypress), w$1(this.input, !0), this.output.on("resize", this.render), this.render(), this.once("submit", () => {
				this.output.write(import_src$1.cursor.show), this.output.off("resize", this.render), w$1(this.input, !1), t(this.value);
			}), this.once("cancel", () => {
				this.output.write(import_src$1.cursor.show), this.output.off("resize", this.render), w$1(this.input, !1), t(C);
			});
		});
	}
	_isActionKey(t, e) {
		return t === "	";
	}
	_setValue(t) {
		this.value = t, this.emit("value", this.value);
	}
	_setUserInput(t, e) {
		this.userInput = t ?? "", this.emit("userInput", this.userInput), e && this._track && this.rl && (this.rl.write(this.userInput), this._cursor = this.rl.cursor);
	}
	_clearUserInput() {
		this.rl?.write(null, {
			ctrl: !0,
			name: "u"
		}), this._setUserInput("");
	}
	onKeypress(t, e) {
		if (this._track && e.name !== "return" && (e.name && this._isActionKey(t, e) && this.rl?.write(null, {
			ctrl: !0,
			name: "h"
		}), this._cursor = this.rl?.cursor ?? 0, this._setUserInput(this.rl?.line)), this.state === "error" && (this.state = "active"), e?.name && (!this._track && u$1.aliases.has(e.name) && this.emit("cursor", u$1.aliases.get(e.name)), u$1.actions.has(e.name) && this.emit("cursor", e.name)), t && (t.toLowerCase() === "y" || t.toLowerCase() === "n") && this.emit("confirm", t.toLowerCase() === "y"), this.emit("key", t?.toLowerCase(), e), e?.name === "return") {
			if (this.opts.validate) {
				const s = this.opts.validate(this.value);
				s && (this.error = s instanceof Error ? s.message : s, this.state = "error", this.rl?.write(this.userInput));
			}
			this.state !== "error" && (this.state = "submit");
		}
		V$1([
			t,
			e?.name,
			e?.sequence
		], "cancel") && (this.state = "cancel"), (this.state === "submit" || this.state === "cancel") && this.emit("finalize"), this.render(), (this.state === "submit" || this.state === "cancel") && this.close();
	}
	close() {
		this.input.unpipe(), this.input.removeListener("keypress", this.onKeypress), this.output.write(`
`), w$1(this.input, !1), this.rl?.close(), this.rl = void 0, this.emit(`${this.state}`, this.value), this.unsubscribe();
	}
	restoreCursor() {
		const t = wrapAnsi(this._prevFrame, process.stdout.columns, {
			hard: !0,
			trim: !1
		}).split(`
`).length - 1;
		this.output.write(import_src$1.cursor.move(-999, t * -1));
	}
	render() {
		const t = wrapAnsi(this._render(this) ?? "", process.stdout.columns, {
			hard: !0,
			trim: !1
		});
		if (t !== this._prevFrame) {
			if (this.state === "initial") this.output.write(import_src$1.cursor.hide);
			else {
				const e = j(this._prevFrame, t), s = A(this.output);
				if (this.restoreCursor(), e) {
					const i = Math.max(0, e.numLinesAfter - s), n = Math.max(0, e.numLinesBefore - s);
					let o = e.lines.find((a) => a >= i);
					if (o === void 0) {
						this._prevFrame = t;
						return;
					}
					if (e.lines.length === 1) {
						this.output.write(import_src$1.cursor.move(0, o - n)), this.output.write(import_src$1.erase.lines(1));
						const a = t.split(`
`);
						this.output.write(a[o]), this._prevFrame = t, this.output.write(import_src$1.cursor.move(0, a.length - o - 1));
						return;
					} else if (e.lines.length > 1) {
						if (i < n) o = i;
						else {
							const h = o - n;
							h > 0 && this.output.write(import_src$1.cursor.move(0, h));
						}
						this.output.write(import_src$1.erase.down());
						const a = t.split(`
`).slice(o);
						this.output.write(a.join(`
`)), this._prevFrame = t;
						return;
					}
				}
				this.output.write(import_src$1.erase.down());
			}
			this.output.write(t), this.state === "initial" && (this.state = "active"), this._prevFrame = t;
		}
	}
};
function W(r, t) {
	if (r === void 0 || t.length === 0) return 0;
	const e = t.findIndex((s) => s.value === r);
	return e !== -1 ? e : 0;
}
function B(r, t) {
	return (t.label ?? String(t.value)).toLowerCase().includes(r.toLowerCase());
}
function J$1(r, t) {
	if (t) return r ? t : t[0];
}
let H$1 = class extends p {
	filteredOptions;
	multiple;
	isNavigating = !1;
	selectedValues = [];
	focusedValue;
	#e = 0;
	#o = "";
	#t;
	#n;
	#a;
	get cursor() {
		return this.#e;
	}
	get userInputWithCursor() {
		if (!this.userInput) return styleText(["inverse", "hidden"], "_");
		if (this._cursor >= this.userInput.length) return `${this.userInput}\u2588`;
		const t = this.userInput.slice(0, this._cursor), [e, ...s] = this.userInput.slice(this._cursor);
		return `${t}${styleText("inverse", e)}${s.join("")}`;
	}
	get options() {
		return typeof this.#n == "function" ? this.#n() : this.#n;
	}
	constructor(t) {
		super(t), this.#n = t.options, this.#a = t.placeholder;
		const e = this.options;
		this.filteredOptions = [...e], this.multiple = t.multiple === !0, this.#t = typeof t.options == "function" ? t.filter : t.filter ?? B;
		let s;
		if (t.initialValue && Array.isArray(t.initialValue) ? this.multiple ? s = t.initialValue : s = t.initialValue.slice(0, 1) : !this.multiple && this.options.length > 0 && (s = [this.options[0].value]), s) for (const i of s) {
			const n = e.findIndex((o) => o.value === i);
			n !== -1 && (this.toggleSelected(i), this.#e = n);
		}
		this.focusedValue = this.options[this.#e]?.value, this.on("key", (i, n) => this.#s(i, n)), this.on("userInput", (i) => this.#i(i));
	}
	_isActionKey(t, e) {
		return t === "	" || this.multiple && this.isNavigating && e.name === "space" && t !== void 0 && t !== "";
	}
	#s(t, e) {
		const s = e.name === "up", i = e.name === "down", n = e.name === "return", o = this.userInput === "" || this.userInput === "	", a = this.#a, h = this.options, l = a !== void 0 && a !== "" && h.some((f) => !f.disabled && (this.#t ? this.#t(a, f) : !0));
		if (e.name === "tab" && o && l) {
			this.userInput === "	" && this._clearUserInput(), this._setUserInput(a, !0), this.isNavigating = !1;
			return;
		}
		s || i ? (this.#e = d$1(this.#e, s ? -1 : 1, this.filteredOptions), this.focusedValue = this.filteredOptions[this.#e]?.value, this.multiple || (this.selectedValues = [this.focusedValue]), this.isNavigating = !0) : n ? this.value = J$1(this.multiple, this.selectedValues) : this.multiple ? this.focusedValue !== void 0 && (e.name === "tab" || this.isNavigating && e.name === "space") ? this.toggleSelected(this.focusedValue) : this.isNavigating = !1 : (this.focusedValue && (this.selectedValues = [this.focusedValue]), this.isNavigating = !1);
	}
	deselectAll() {
		this.selectedValues = [];
	}
	toggleSelected(t) {
		this.filteredOptions.length !== 0 && (this.multiple ? this.selectedValues.includes(t) ? this.selectedValues = this.selectedValues.filter((e) => e !== t) : this.selectedValues = [...this.selectedValues, t] : this.selectedValues = [t]);
	}
	#i(t) {
		if (t !== this.#o) {
			this.#o = t;
			const e = this.options;
			t && this.#t ? this.filteredOptions = e.filter((n) => this.#t?.(t, n)) : this.filteredOptions = [...e];
			const s = W(this.focusedValue, this.filteredOptions);
			this.#e = d$1(s, 0, this.filteredOptions);
			const i = this.filteredOptions[this.#e];
			i && !i.disabled ? this.focusedValue = i.value : this.focusedValue = void 0, this.multiple || (this.focusedValue !== void 0 ? this.toggleSelected(this.focusedValue) : this.deselectAll());
		}
	}
};
var Q$1 = class extends p {
	get cursor() {
		return this.value ? 0 : 1;
	}
	get _value() {
		return this.cursor === 0;
	}
	constructor(t) {
		super(t, !1), this.value = !!t.initialValue, this.on("userInput", () => {
			this.value = this._value;
		}), this.on("confirm", (e) => {
			this.output.write(import_src$1.cursor.move(0, -1)), this.value = e, this.state = "submit", this.close();
		}), this.on("cursor", () => {
			this.value = !this.value;
		});
	}
};
var nt$1 = class extends p {
	options;
	cursor = 0;
	get _selectedValue() {
		return this.options[this.cursor];
	}
	changeValue() {
		this.value = this._selectedValue.value;
	}
	constructor(t) {
		super(t, !1), this.options = t.options;
		const e = this.options.findIndex(({ value: i }) => i === t.initialValue), s = e === -1 ? 0 : e;
		this.cursor = this.options[s].disabled ? d$1(s, 1, this.options) : s, this.changeValue(), this.on("cursor", (i) => {
			switch (i) {
				case "left":
				case "up":
					this.cursor = d$1(this.cursor, -1, this.options);
					break;
				case "down":
				case "right":
					this.cursor = d$1(this.cursor, 1, this.options);
					break;
			}
			this.changeValue();
		});
	}
};
var at$1 = class extends p {
	get userInputWithCursor() {
		if (this.state === "submit") return this.userInput;
		const t = this.userInput;
		if (this.cursor >= t.length) return `${this.userInput}\u2588`;
		const e = t.slice(0, this.cursor), [s, ...i] = t.slice(this.cursor);
		return `${e}${styleText("inverse", s)}${i.join("")}`;
	}
	get cursor() {
		return this._cursor;
	}
	constructor(t) {
		super({
			...t,
			initialUserInput: t.initialUserInput ?? t.initialValue
		}), this.on("userInput", (e) => {
			this._setValue(e);
		}), this.on("finalize", () => {
			this.value || (this.value = t.defaultValue), this.value === void 0 && (this.value = "");
		});
	}
};
//#endregion
//#region ../../node_modules/.pnpm/@clack+prompts@1.2.0/node_modules/@clack/prompts/dist/index.mjs
function Ze() {
	return process$1.platform !== "win32" ? process$1.env.TERM !== "linux" : !!process$1.env.CI || !!process$1.env.WT_SESSION || !!process$1.env.TERMINUS_SUBLIME || process$1.env.ConEmuTask === "{cmd::Cmder}" || process$1.env.TERM_PROGRAM === "Terminus-Sublime" || process$1.env.TERM_PROGRAM === "vscode" || process$1.env.TERM === "xterm-256color" || process$1.env.TERM === "alacritty" || process$1.env.TERMINAL_EMULATOR === "JetBrains-JediTerm";
}
const ee = Ze(), ae = () => process.env.CI === "true", Te = (e) => e.isTTY === !0, w = (e, i) => ee ? e : i, _e = w("◆", "*"), oe = w("■", "x"), ue = w("▲", "x"), F = w("◇", "o"), le = w("┌", "T"), d = w("│", "|"), E = w("└", "—"), Ie = w("┐", "T"), Ee = w("┘", "—"), z = w("●", ">"), H = w("○", " ");
w("◻", "[•]");
const U = w("◼", "[+]"), J = w("◻", "[ ]");
w("▪", "•");
const se = w("─", "-"), ce = w("╮", "+"), Ge = w("├", "+"), $e = w("╯", "+"), de = w("╰", "+"), Oe = w("╭", "+"), he = w("●", "•"), pe = w("◆", "*"), me = w("▲", "!"), ge = w("■", "x"), V = (e) => {
	switch (e) {
		case "initial":
		case "active": return styleText("cyan", _e);
		case "cancel": return styleText("red", oe);
		case "error": return styleText("yellow", ue);
		case "submit": return styleText("green", F);
	}
}, ye = (e) => {
	switch (e) {
		case "initial":
		case "active": return styleText("cyan", d);
		case "cancel": return styleText("red", d);
		case "error": return styleText("yellow", d);
		case "submit": return styleText("green", d);
	}
}, et = (e, i, s, r, u) => {
	let n = i, o = 0;
	for (let c = s; c < r; c++) {
		const a = e[c];
		if (n = n - a.length, o++, n <= u) break;
	}
	return {
		lineCount: n,
		removals: o
	};
}, Y = ({ cursor: e, options: i, style: s, output: r = process.stdout, maxItems: u = Number.POSITIVE_INFINITY, columnPadding: n = 0, rowPadding: o = 4 }) => {
	const c = O$1(r) - n, a = A(r), l = styleText("dim", "..."), $ = Math.max(a - o, 0), y = Math.max(Math.min(u, $), 5);
	let p = 0;
	e >= y - 3 && (p = Math.max(Math.min(e - y + 3, i.length - y), 0));
	let m = y < i.length && p > 0, g = y < i.length && p + y < i.length;
	const S = Math.min(p + y, i.length), h = [];
	let f = 0;
	m && f++, g && f++;
	const v = p + (m ? 1 : 0), T = S - (g ? 1 : 0);
	for (let b = v; b < T; b++) {
		const x = wrapAnsi(s(i[b], b === e), c, {
			hard: !0,
			trim: !1
		}).split(`
`);
		h.push(x), f += x.length;
	}
	if (f > $) {
		let b = 0, x = 0, G = f;
		const M = e - v, R = (j, D) => et(h, G, j, D, $);
		m ? ({lineCount: G, removals: b} = R(0, M), G > $ && ({lineCount: G, removals: x} = R(M + 1, h.length))) : ({lineCount: G, removals: x} = R(M + 1, h.length), G > $ && ({lineCount: G, removals: b} = R(0, M))), b > 0 && (m = !0, h.splice(0, b)), x > 0 && (g = !0, h.splice(h.length - x, x));
	}
	const C = [];
	m && C.push(l);
	for (const b of h) for (const x of b) C.push(x);
	return g && C.push(l), C;
};
function Re(e, i) {
	if (!e) return !0;
	const s = (i.label ?? String(i.value ?? "")).toLowerCase(), r = (i.hint ?? "").toLowerCase(), u = String(i.value).toLowerCase(), n = e.toLowerCase();
	return s.includes(n) || r.includes(n) || u.includes(n);
}
const st = (e) => {
	const i = (r, u, n, o) => {
		const c = n.includes(r.value), a = r.label ?? String(r.value ?? ""), l = r.hint && o !== void 0 && r.value === o ? styleText("dim", ` (${r.hint})`) : "", $ = c ? styleText("green", U) : styleText("dim", J);
		return r.disabled ? `${styleText("gray", J)} ${styleText(["strikethrough", "gray"], a)}` : u ? `${$} ${a}${l}` : `${$} ${styleText("dim", a)}`;
	}, s = new H$1({
		options: e.options,
		multiple: !0,
		placeholder: e.placeholder,
		filter: e.filter ?? ((r, u) => Re(r, u)),
		validate: () => {
			if (e.required && s.selectedValues.length === 0) return "Please select at least one item";
		},
		initialValue: e.initialValues,
		signal: e.signal,
		input: e.input,
		output: e.output,
		render() {
			const r = e.withGuide ?? u$1.withGuide, u = `${r ? `${styleText("gray", d)}
` : ""}${V(this.state)}  ${e.message}
`, n = this.userInput, o = e.placeholder, c = n === "" && o !== void 0, a = this.isNavigating || c ? styleText("dim", c ? o : n) : this.userInputWithCursor, l = this.options, $ = this.filteredOptions.length !== l.length ? styleText("dim", ` (${this.filteredOptions.length} match${this.filteredOptions.length === 1 ? "" : "es"})`) : "";
			switch (this.state) {
				case "submit": return `${u}${r ? `${styleText("gray", d)}  ` : ""}${styleText("dim", `${this.selectedValues.length} items selected`)}`;
				case "cancel": return `${u}${r ? `${styleText("gray", d)}  ` : ""}${styleText(["strikethrough", "dim"], n)}`;
				default: {
					const y = this.state === "error" ? "yellow" : "cyan", p = r ? `${styleText(y, d)}  ` : "", m = r ? styleText(y, E) : "", g = [
						`${styleText("dim", "↑/↓")} to navigate`,
						`${styleText("dim", this.isNavigating ? "Space/Tab:" : "Tab:")} select`,
						`${styleText("dim", "Enter:")} confirm`,
						`${styleText("dim", "Type:")} to search`
					], S = this.filteredOptions.length === 0 && n ? [`${p}${styleText("yellow", "No matches found")}`] : [], h = this.state === "error" ? [`${p}${styleText("yellow", this.error)}`] : [], f = [
						...`${u}${r ? styleText(y, d) : ""}`.split(`
`),
						`${p}${styleText("dim", "Search:")} ${a}${$}`,
						...S,
						...h
					], v = [`${p}${g.join(" • ")}`, m], T = Y({
						cursor: this.cursor,
						options: this.filteredOptions,
						style: (C, b) => i(C, b, this.selectedValues, this.focusedValue),
						maxItems: e.maxItems,
						output: e.output,
						rowPadding: f.length + v.length
					});
					return [
						...f,
						...T.map((C) => `${p}${C}`),
						...v
					].join(`
`);
				}
			}
		}
	});
	return s.prompt();
}, rt = [
	Oe,
	ce,
	de,
	$e
], it = [
	le,
	Ie,
	E,
	Ee
];
function Pe(e, i, s, r) {
	let u = s, n = s;
	return r === "center" ? u = Math.floor((i - e) / 2) : r === "right" && (u = i - e - s), n = i - u - e, [u, n];
}
const nt = (e) => e, at = (e = "", i = "", s) => {
	const r = s?.output ?? process.stdout, u = O$1(r), n = 2, o = s?.titlePadding ?? 1, c = s?.contentPadding ?? 2, a = s?.width === void 0 || s.width === "auto" ? 1 : Math.min(1, s.width), l = s?.withGuide ?? u$1.withGuide ? `${d} ` : "", $ = s?.formatBorder ?? nt, y = (s?.rounded ? rt : it).map($), p = $(se), m = $(d), g = fastStringWidth(l), S = fastStringWidth(i), h = u - g;
	let f = Math.floor(u * a) - g;
	if (s?.width === "auto") {
		const R = e.split(`
`);
		let j = S + o * 2;
		for (const ie of R) {
			const W = fastStringWidth(ie) + c * 2;
			W > j && (j = W);
		}
		const D = j + n;
		D < f && (f = D);
	}
	f % 2 !== 0 && (f < h ? f++ : f--);
	const v = f - n, T = v - o * 2, C = S > T ? `${i.slice(0, T - 3)}...` : i, [b, x] = Pe(fastStringWidth(C), v, o, s?.titleAlign), G = wrapAnsi(e, v - c * 2, {
		hard: !0,
		trim: !1
	});
	r.write(`${l}${y[0]}${p.repeat(b)}${C}${p.repeat(x)}${y[1]}
`);
	const M = G.split(`
`);
	for (const R of M) {
		const [j, D] = Pe(fastStringWidth(R), v, c, s?.contentAlign);
		r.write(`${l}${m}${" ".repeat(j)}${R}${" ".repeat(D)}${m}
`);
	}
	r.write(`${l}${y[2]}${p.repeat(v)}${y[3]}
`);
}, ot = (e) => {
	const i = e.active ?? "Yes", s = e.inactive ?? "No";
	return new Q$1({
		active: i,
		inactive: s,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValue: e.initialValue ?? !0,
		render() {
			const r = e.withGuide ?? u$1.withGuide, u = `${V(this.state)}  `, n = r ? `${styleText("gray", d)}  ` : "", o = R(e.output, e.message, n, u), c = `${r ? `${styleText("gray", d)}
` : ""}${o}
`, a = this.value ? i : s;
			switch (this.state) {
				case "submit": return `${c}${r ? `${styleText("gray", d)}  ` : ""}${styleText("dim", a)}`;
				case "cancel": return `${c}${r ? `${styleText("gray", d)}  ` : ""}${styleText(["strikethrough", "dim"], a)}${r ? `
${styleText("gray", d)}` : ""}`;
				default: {
					const l = r ? `${styleText("cyan", d)}  ` : "", $ = r ? styleText("cyan", E) : "";
					return `${c}${l}${this.value ? `${styleText("green", z)} ${i}` : `${styleText("dim", H)} ${styleText("dim", i)}`}${e.vertical ? r ? `
${styleText("cyan", d)}  ` : `
` : ` ${styleText("dim", "/")} `}${this.value ? `${styleText("dim", H)} ${styleText("dim", s)}` : `${styleText("green", z)} ${s}`}
${$}
`;
				}
			}
		}
	}).prompt();
}, O = {
	message: (e = [], { symbol: i = styleText("gray", d), secondarySymbol: s = styleText("gray", d), output: r = process.stdout, spacing: u = 1, withGuide: n } = {}) => {
		const o = [], c = n ?? u$1.withGuide, a = c ? s : "", l = c ? `${i}  ` : "", $ = c ? `${s}  ` : "";
		for (let p = 0; p < u; p++) o.push(a);
		const y = Array.isArray(e) ? e : e.split(`
`);
		if (y.length > 0) {
			const [p, ...m] = y;
			p.length > 0 ? o.push(`${l}${p}`) : o.push(c ? i : "");
			for (const g of m) g.length > 0 ? o.push(`${$}${g}`) : o.push(c ? s : "");
		}
		r.write(`${o.join(`
`)}
`);
	},
	info: (e, i) => {
		O.message(e, {
			...i,
			symbol: styleText("blue", he)
		});
	},
	success: (e, i) => {
		O.message(e, {
			...i,
			symbol: styleText("green", pe)
		});
	},
	step: (e, i) => {
		O.message(e, {
			...i,
			symbol: styleText("green", F)
		});
	},
	warn: (e, i) => {
		O.message(e, {
			...i,
			symbol: styleText("yellow", me)
		});
	},
	warning: (e, i) => {
		O.warn(e, i);
	},
	error: (e, i) => {
		O.message(e, {
			...i,
			symbol: styleText("red", ge)
		});
	}
}, pt = (e = "", i) => {
	const s = i?.output ?? process.stdout, r = i?.withGuide ?? u$1.withGuide ? `${styleText("gray", E)}  ` : "";
	s.write(`${r}${styleText("red", e)}

`);
}, mt = (e = "", i) => {
	const s = i?.output ?? process.stdout, r = i?.withGuide ?? u$1.withGuide ? `${styleText("gray", le)}  ` : "";
	s.write(`${r}${e}
`);
}, gt = (e = "", i) => {
	const s = i?.output ?? process.stdout, r = i?.withGuide ?? u$1.withGuide ? `${styleText("gray", d)}
${styleText("gray", E)}  ` : "";
	s.write(`${r}${e}

`);
}, ft = (e) => styleText("dim", e), vt = (e, i, s) => {
	const r = {
		hard: !0,
		trim: !1
	}, u = wrapAnsi(e, i, r).split(`
`), n = u.reduce((a, l) => Math.max(fastStringWidth(l), a), 0);
	return wrapAnsi(e, i - (u.map(s).reduce((a, l) => Math.max(fastStringWidth(l), a), 0) - n), r);
}, wt = (e = "", i = "", s) => {
	const r = s?.output ?? process$1.stdout, u = s?.withGuide ?? u$1.withGuide, n = s?.format ?? ft, o = [
		"",
		...vt(e, O$1(r) - 6, n).split(`
`).map(n),
		""
	], c = fastStringWidth(i), a = Math.max(o.reduce((p, m) => {
		const g = fastStringWidth(m);
		return g > p ? g : p;
	}, 0), c) + 2, l = o.map((p) => `${styleText("gray", d)}  ${p}${" ".repeat(a - fastStringWidth(p))}${styleText("gray", d)}`).join(`
`), $ = u ? `${styleText("gray", d)}
` : "", y = u ? Ge : de;
	r.write(`${$}${styleText("green", F)}  ${styleText("reset", i)} ${styleText("gray", se.repeat(Math.max(a - c - 1, 1)) + ce)}
${l}
${styleText("gray", y + se.repeat(a + 2) + $e)}
`);
}, Ct = (e) => styleText("magenta", e), fe = ({ indicator: e = "dots", onCancel: i, output: s = process.stdout, cancelMessage: r, errorMessage: u, frames: n = ee ? [
	"◒",
	"◐",
	"◓",
	"◑"
] : [
	"•",
	"o",
	"O",
	"0"
], delay: o = ee ? 80 : 120, signal: c, ...a } = {}) => {
	const l = ae();
	let $, y, p = !1, m = !1, g = "", S, h = performance.now();
	const f = O$1(s), v = a?.styleFrame ?? Ct, T = (_) => {
		const A = _ > 1 ? u ?? u$1.messages.error : r ?? u$1.messages.cancel;
		m = _ === 1, p && (W(A, _), m && typeof i == "function" && i());
	}, C = () => T(2), b = () => T(1), x = () => {
		process.on("uncaughtExceptionMonitor", C), process.on("unhandledRejection", C), process.on("SIGINT", b), process.on("SIGTERM", b), process.on("exit", T), c && c.addEventListener("abort", b);
	}, G = () => {
		process.removeListener("uncaughtExceptionMonitor", C), process.removeListener("unhandledRejection", C), process.removeListener("SIGINT", b), process.removeListener("SIGTERM", b), process.removeListener("exit", T), c && c.removeEventListener("abort", b);
	}, M = () => {
		if (S === void 0) return;
		l && s.write(`
`);
		const _ = wrapAnsi(S, f, {
			hard: !0,
			trim: !1
		}).split(`
`);
		_.length > 1 && s.write(import_src$1.cursor.up(_.length - 1)), s.write(import_src$1.cursor.to(0)), s.write(import_src$1.erase.down());
	}, R = (_) => _.replace(/\.+$/, ""), j = (_) => {
		const A = (performance.now() - _) / 1e3, k = Math.floor(A / 60), L = Math.floor(A % 60);
		return k > 0 ? `[${k}m ${L}s]` : `[${L}s]`;
	}, D = a.withGuide ?? u$1.withGuide, ie = (_ = "") => {
		p = !0, $ = z$1({ output: s }), g = R(_), h = performance.now(), D && s.write(`${styleText("gray", d)}
`);
		let A = 0, k = 0;
		x(), y = setInterval(() => {
			if (l && g === S) return;
			M(), S = g;
			const L = v(n[A]);
			let Z;
			if (l) Z = `${L}  ${g}...`;
			else if (e === "timer") Z = `${L}  ${g} ${j(h)}`;
			else {
				const Be = ".".repeat(Math.floor(k)).slice(0, 3);
				Z = `${L}  ${g}${Be}`;
			}
			const Ne = wrapAnsi(Z, f, {
				hard: !0,
				trim: !1
			});
			s.write(Ne), A = A + 1 < n.length ? A + 1 : 0, k = k < 4 ? k + .125 : 0;
		}, o);
	}, W = (_ = "", A = 0, k = !1) => {
		if (!p) return;
		p = !1, clearInterval(y), M();
		const L = A === 0 ? styleText("green", F) : A === 1 ? styleText("red", oe) : styleText("red", ue);
		g = _ ?? g, k || (e === "timer" ? s.write(`${L}  ${g} ${j(h)}
`) : s.write(`${L}  ${g}
`)), G(), $();
	};
	return {
		start: ie,
		stop: (_ = "") => W(_, 0),
		message: (_ = "") => {
			g = R(_ ?? g);
		},
		cancel: (_ = "") => W(_, 1),
		error: (_ = "") => W(_, 2),
		clear: () => W("", 0, !0),
		get isCancelled() {
			return m;
		}
	};
};
w("─", "-"), w("━", "="), w("█", "#");
const re = (e, i) => e.includes(`
`) ? e.split(`
`).map((s) => i(s)).join(`
`) : i(e), _t = (e) => {
	const i = (s, r) => {
		const u = s.label ?? String(s.value);
		switch (r) {
			case "disabled": return `${styleText("gray", H)} ${re(u, (n) => styleText("gray", n))}${s.hint ? ` ${styleText("dim", `(${s.hint ?? "disabled"})`)}` : ""}`;
			case "selected": return `${re(u, (n) => styleText("dim", n))}`;
			case "active": return `${styleText("green", z)} ${u}${s.hint ? ` ${styleText("dim", `(${s.hint})`)}` : ""}`;
			case "cancelled": return `${re(u, (n) => styleText(["strikethrough", "dim"], n))}`;
			default: return `${styleText("dim", H)} ${re(u, (n) => styleText("dim", n))}`;
		}
	};
	return new nt$1({
		options: e.options,
		signal: e.signal,
		input: e.input,
		output: e.output,
		initialValue: e.initialValue,
		render() {
			const s = e.withGuide ?? u$1.withGuide, r = `${V(this.state)}  `, u = `${ye(this.state)}  `, n = R(e.output, e.message, u, r), o = `${s ? `${styleText("gray", d)}
` : ""}${n}
`;
			switch (this.state) {
				case "submit": {
					const c = s ? `${styleText("gray", d)}  ` : "";
					return `${o}${R(e.output, i(this.options[this.cursor], "selected"), c)}`;
				}
				case "cancel": {
					const c = s ? `${styleText("gray", d)}  ` : "";
					return `${o}${R(e.output, i(this.options[this.cursor], "cancelled"), c)}${s ? `
${styleText("gray", d)}` : ""}`;
				}
				default: {
					const c = s ? `${styleText("cyan", d)}  ` : "", a = s ? styleText("cyan", E) : "", l = o.split(`
`).length, $ = s ? 2 : 1;
					return `${o}${c}${Y({
						output: e.output,
						cursor: this.cursor,
						options: this.options,
						maxItems: e.maxItems,
						columnPadding: c.length,
						rowPadding: l + $,
						style: (y, p) => i(y, y.disabled ? "disabled" : p ? "active" : "inactive")
					}).join(`
${c}`)}
${a}
`;
				}
			}
		}
	}).prompt();
}, je = `${styleText("gray", d)}  `, K = {
	message: async (e, { symbol: i = styleText("gray", d) } = {}) => {
		process.stdout.write(`${styleText("gray", d)}
${i}  `);
		let s = 3;
		for await (let r of e) {
			r = r.replace(/\n/g, `
${je}`), r.includes(`
`) && (s = 3 + stripVTControlCharacters(r.slice(r.lastIndexOf(`
`))).length);
			const u = stripVTControlCharacters(r).length;
			s + u < process.stdout.columns ? (s += u, process.stdout.write(r)) : (process.stdout.write(`
${je}${r.trimStart()}`), s = 3 + stripVTControlCharacters(r.trimStart()).length);
		}
		process.stdout.write(`
`);
	},
	info: (e) => K.message(e, { symbol: styleText("blue", he) }),
	success: (e) => K.message(e, { symbol: styleText("green", pe) }),
	step: (e) => K.message(e, { symbol: styleText("green", F) }),
	warn: (e) => K.message(e, { symbol: styleText("yellow", me) }),
	warning: (e) => K.warn(e),
	error: (e) => K.message(e, { symbol: styleText("red", ge) })
}, Et = async (e, i) => {
	for (const s of e) {
		if (s.enabled === !1) continue;
		const r = fe(i);
		r.start(s.title);
		const u = await s.task(r.message);
		r.stop(u || s.title);
	}
}, xt = (e) => e.replace(/\x1b\[(?:\d+;)*\d*[ABCDEFGHfJKSTsu]|\x1b\[(s|u)/g, ""), Gt = (e) => {
	const i = e.output ?? process.stdout, s = O$1(i), r = styleText("gray", d), u = e.spacing ?? 1, n = 3, o = e.retainLog === !0, c = !ae() && Te(i);
	i.write(`${r}
`), i.write(`${styleText("green", F)}  ${e.title}
`);
	for (let h = 0; h < u; h++) i.write(`${r}
`);
	const a = [{
		value: "",
		full: ""
	}];
	let l = !1;
	const $ = (h) => {
		if (a.length === 0) return;
		let f = 0;
		h && (f += u + 2);
		for (const v of a) {
			const { value: T, result: C } = v;
			let b = C?.message ?? T;
			if (b.length === 0) continue;
			C === void 0 && v.header !== void 0 && v.header !== "" && (b += `
${v.header}`);
			const x = b.split(`
`).reduce((G, M) => M === "" ? G + 1 : G + Math.ceil((M.length + n) / s), 0);
			f += x;
		}
		f > 0 && (f += 1, i.write(import_src$1.erase.lines(f)));
	}, y = (h, f, v) => {
		const T = v ? `${h.full}
${h.value}` : h.value;
		h.header !== void 0 && h.header !== "" && O.message(h.header.split(`
`).map((C) => styleText("bold", C)), {
			output: i,
			secondarySymbol: r,
			symbol: r,
			spacing: 0
		}), O.message(T.split(`
`).map((C) => styleText("dim", C)), {
			output: i,
			secondarySymbol: r,
			symbol: r,
			spacing: f ?? u
		});
	}, p = () => {
		for (const h of a) {
			const { header: f, value: v, full: T } = h;
			(f === void 0 || f.length === 0) && v.length === 0 || y(h, void 0, o === !0 && T.length > 0);
		}
	}, m = (h, f, v) => {
		if ($(!1), (v?.raw !== !0 || !l) && h.value !== "" && (h.value += `
`), h.value += xt(f), l = v?.raw === !0, e.limit !== void 0) {
			const T = h.value.split(`
`), C = T.length - e.limit;
			if (C > 0) {
				const b = T.splice(0, C);
				o && (h.full += (h.full === "" ? "" : `
`) + b.join(`
`));
			}
			h.value = T.join(`
`);
		}
		c && g();
	}, g = () => {
		for (const h of a) h.result ? h.result.status === "error" ? O.error(h.result.message, {
			output: i,
			secondarySymbol: r,
			spacing: 0
		}) : O.success(h.result.message, {
			output: i,
			secondarySymbol: r,
			spacing: 0
		}) : h.value !== "" && y(h, 0);
	}, S = (h, f) => {
		$(!1), h.result = f, c && g();
	};
	return {
		message(h, f) {
			m(a[0], h, f);
		},
		group(h) {
			const f = {
				header: h,
				value: "",
				full: ""
			};
			return a.push(f), {
				message(v, T) {
					m(f, v, T);
				},
				error(v) {
					S(f, {
						status: "error",
						message: v
					});
				},
				success(v) {
					S(f, {
						status: "success",
						message: v
					});
				}
			};
		},
		error(h, f) {
			$(!0), O.error(h, {
				output: i,
				secondarySymbol: r,
				spacing: 1
			}), f?.showLog !== !1 && p(), a.splice(1, a.length - 1), a[0].value = "", a[0].full = "";
		},
		success(h, f) {
			$(!0), O.success(h, {
				output: i,
				secondarySymbol: r,
				spacing: 1
			}), f?.showLog === !0 && p(), a.splice(1, a.length - 1), a[0].value = "", a[0].full = "";
		}
	};
}, Ot = (e) => new at$1({
	validate: e.validate,
	placeholder: e.placeholder,
	defaultValue: e.defaultValue,
	initialValue: e.initialValue,
	output: e.output,
	signal: e.signal,
	input: e.input,
	render() {
		const i = e?.withGuide ?? u$1.withGuide, s = `${`${i ? `${styleText("gray", d)}
` : ""}${V(this.state)}  `}${e.message}
`, r = e.placeholder ? styleText("inverse", e.placeholder[0]) + styleText("dim", e.placeholder.slice(1)) : styleText(["inverse", "hidden"], "_"), u = this.userInput ? this.userInputWithCursor : r, n = this.value ?? "";
		switch (this.state) {
			case "error": {
				const o = this.error ? `  ${styleText("yellow", this.error)}` : "", c = i ? `${styleText("yellow", d)}  ` : "", a = i ? styleText("yellow", E) : "";
				return `${s.trim()}
${c}${u}
${a}${o}
`;
			}
			case "submit": {
				const o = n ? `  ${styleText("dim", n)}` : "";
				return `${s}${i ? styleText("gray", d) : ""}${o}`;
			}
			case "cancel": {
				const o = n ? `  ${styleText(["strikethrough", "dim"], n)}` : "", c = i ? styleText("gray", d) : "";
				return `${s}${c}${o}${n.trim() ? `
${c}` : ""}`;
			}
			default: return `${s}${i ? `${styleText("cyan", d)}  ` : ""}${u}
${i ? styleText("cyan", E) : ""}
`;
		}
	}
}).prompt();
//#endregion
//#region ../../node_modules/.pnpm/ms@2.1.3/node_modules/ms/index.js
var require_ms = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* Helpers.
	*/
	var s = 1e3;
	var m = s * 60;
	var h = m * 60;
	var d = h * 24;
	var w = d * 7;
	var y = d * 365.25;
	/**
	* Parse or format the given `val`.
	*
	* Options:
	*
	*  - `long` verbose formatting [false]
	*
	* @param {String|Number} val
	* @param {Object} [options]
	* @throws {Error} throw an error if val is not a non-empty string or a number
	* @return {String|Number}
	* @api public
	*/
	module.exports = function(val, options) {
		options = options || {};
		var type = typeof val;
		if (type === "string" && val.length > 0) return parse(val);
		else if (type === "number" && isFinite(val)) return options.long ? fmtLong(val) : fmtShort(val);
		throw new Error("val is not a non-empty string or a valid number. val=" + JSON.stringify(val));
	};
	/**
	* Parse the given `str` and return milliseconds.
	*
	* @param {String} str
	* @return {Number}
	* @api private
	*/
	function parse(str) {
		str = String(str);
		if (str.length > 100) return;
		var match = /^(-?(?:\d+)?\.?\d+) *(milliseconds?|msecs?|ms|seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|weeks?|w|years?|yrs?|y)?$/i.exec(str);
		if (!match) return;
		var n = parseFloat(match[1]);
		switch ((match[2] || "ms").toLowerCase()) {
			case "years":
			case "year":
			case "yrs":
			case "yr":
			case "y": return n * y;
			case "weeks":
			case "week":
			case "w": return n * w;
			case "days":
			case "day":
			case "d": return n * d;
			case "hours":
			case "hour":
			case "hrs":
			case "hr":
			case "h": return n * h;
			case "minutes":
			case "minute":
			case "mins":
			case "min":
			case "m": return n * m;
			case "seconds":
			case "second":
			case "secs":
			case "sec":
			case "s": return n * s;
			case "milliseconds":
			case "millisecond":
			case "msecs":
			case "msec":
			case "ms": return n;
			default: return;
		}
	}
	/**
	* Short format for `ms`.
	*
	* @param {Number} ms
	* @return {String}
	* @api private
	*/
	function fmtShort(ms) {
		var msAbs = Math.abs(ms);
		if (msAbs >= d) return Math.round(ms / d) + "d";
		if (msAbs >= h) return Math.round(ms / h) + "h";
		if (msAbs >= m) return Math.round(ms / m) + "m";
		if (msAbs >= s) return Math.round(ms / s) + "s";
		return ms + "ms";
	}
	/**
	* Long format for `ms`.
	*
	* @param {Number} ms
	* @return {String}
	* @api private
	*/
	function fmtLong(ms) {
		var msAbs = Math.abs(ms);
		if (msAbs >= d) return plural(ms, msAbs, d, "day");
		if (msAbs >= h) return plural(ms, msAbs, h, "hour");
		if (msAbs >= m) return plural(ms, msAbs, m, "minute");
		if (msAbs >= s) return plural(ms, msAbs, s, "second");
		return ms + " ms";
	}
	/**
	* Pluralization helper.
	*/
	function plural(ms, msAbs, n, name) {
		var isPlural = msAbs >= n * 1.5;
		return Math.round(ms / n) + " " + name + (isPlural ? "s" : "");
	}
}));
//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/common.js
var require_common = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* This is the common logic for both the Node.js and web browser
	* implementations of `debug()`.
	*/
	function setup(env) {
		createDebug.debug = createDebug;
		createDebug.default = createDebug;
		createDebug.coerce = coerce;
		createDebug.disable = disable;
		createDebug.enable = enable;
		createDebug.enabled = enabled;
		createDebug.humanize = require_ms();
		createDebug.destroy = destroy;
		Object.keys(env).forEach((key) => {
			createDebug[key] = env[key];
		});
		/**
		* The currently active debug mode names, and names to skip.
		*/
		createDebug.names = [];
		createDebug.skips = [];
		/**
		* Map of special "%n" handling functions, for the debug "format" argument.
		*
		* Valid key names are a single, lower or upper-case letter, i.e. "n" and "N".
		*/
		createDebug.formatters = {};
		/**
		* Selects a color for a debug namespace
		* @param {String} namespace The namespace string for the debug instance to be colored
		* @return {Number|String} An ANSI color code for the given namespace
		* @api private
		*/
		function selectColor(namespace) {
			let hash = 0;
			for (let i = 0; i < namespace.length; i++) {
				hash = (hash << 5) - hash + namespace.charCodeAt(i);
				hash |= 0;
			}
			return createDebug.colors[Math.abs(hash) % createDebug.colors.length];
		}
		createDebug.selectColor = selectColor;
		/**
		* Create a debugger with the given `namespace`.
		*
		* @param {String} namespace
		* @return {Function}
		* @api public
		*/
		function createDebug(namespace) {
			let prevTime;
			let enableOverride = null;
			let namespacesCache;
			let enabledCache;
			function debug(...args) {
				if (!debug.enabled) return;
				const self = debug;
				const curr = Number(/* @__PURE__ */ new Date());
				self.diff = curr - (prevTime || curr);
				self.prev = prevTime;
				self.curr = curr;
				prevTime = curr;
				args[0] = createDebug.coerce(args[0]);
				if (typeof args[0] !== "string") args.unshift("%O");
				let index = 0;
				args[0] = args[0].replace(/%([a-zA-Z%])/g, (match, format) => {
					if (match === "%%") return "%";
					index++;
					const formatter = createDebug.formatters[format];
					if (typeof formatter === "function") {
						const val = args[index];
						match = formatter.call(self, val);
						args.splice(index, 1);
						index--;
					}
					return match;
				});
				createDebug.formatArgs.call(self, args);
				(self.log || createDebug.log).apply(self, args);
			}
			debug.namespace = namespace;
			debug.useColors = createDebug.useColors();
			debug.color = createDebug.selectColor(namespace);
			debug.extend = extend;
			debug.destroy = createDebug.destroy;
			Object.defineProperty(debug, "enabled", {
				enumerable: true,
				configurable: false,
				get: () => {
					if (enableOverride !== null) return enableOverride;
					if (namespacesCache !== createDebug.namespaces) {
						namespacesCache = createDebug.namespaces;
						enabledCache = createDebug.enabled(namespace);
					}
					return enabledCache;
				},
				set: (v) => {
					enableOverride = v;
				}
			});
			if (typeof createDebug.init === "function") createDebug.init(debug);
			return debug;
		}
		function extend(namespace, delimiter) {
			const newDebug = createDebug(this.namespace + (typeof delimiter === "undefined" ? ":" : delimiter) + namespace);
			newDebug.log = this.log;
			return newDebug;
		}
		/**
		* Enables a debug mode by namespaces. This can include modes
		* separated by a colon and wildcards.
		*
		* @param {String} namespaces
		* @api public
		*/
		function enable(namespaces) {
			createDebug.save(namespaces);
			createDebug.namespaces = namespaces;
			createDebug.names = [];
			createDebug.skips = [];
			const split = (typeof namespaces === "string" ? namespaces : "").trim().replace(/\s+/g, ",").split(",").filter(Boolean);
			for (const ns of split) if (ns[0] === "-") createDebug.skips.push(ns.slice(1));
			else createDebug.names.push(ns);
		}
		/**
		* Checks if the given string matches a namespace template, honoring
		* asterisks as wildcards.
		*
		* @param {String} search
		* @param {String} template
		* @return {Boolean}
		*/
		function matchesTemplate(search, template) {
			let searchIndex = 0;
			let templateIndex = 0;
			let starIndex = -1;
			let matchIndex = 0;
			while (searchIndex < search.length) if (templateIndex < template.length && (template[templateIndex] === search[searchIndex] || template[templateIndex] === "*")) if (template[templateIndex] === "*") {
				starIndex = templateIndex;
				matchIndex = searchIndex;
				templateIndex++;
			} else {
				searchIndex++;
				templateIndex++;
			}
			else if (starIndex !== -1) {
				templateIndex = starIndex + 1;
				matchIndex++;
				searchIndex = matchIndex;
			} else return false;
			while (templateIndex < template.length && template[templateIndex] === "*") templateIndex++;
			return templateIndex === template.length;
		}
		/**
		* Disable debug output.
		*
		* @return {String} namespaces
		* @api public
		*/
		function disable() {
			const namespaces = [...createDebug.names, ...createDebug.skips.map((namespace) => "-" + namespace)].join(",");
			createDebug.enable("");
			return namespaces;
		}
		/**
		* Returns true if the given mode name is enabled, false otherwise.
		*
		* @param {String} name
		* @return {Boolean}
		* @api public
		*/
		function enabled(name) {
			for (const skip of createDebug.skips) if (matchesTemplate(name, skip)) return false;
			for (const ns of createDebug.names) if (matchesTemplate(name, ns)) return true;
			return false;
		}
		/**
		* Coerce `val`.
		*
		* @param {Mixed} val
		* @return {Mixed}
		* @api private
		*/
		function coerce(val) {
			if (val instanceof Error) return val.stack || val.message;
			return val;
		}
		/**
		* XXX DO NOT USE. This is a temporary stub function.
		* XXX It WILL be removed in the next major release.
		*/
		function destroy() {
			console.warn("Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
		}
		createDebug.enable(createDebug.load());
		return createDebug;
	}
	module.exports = setup;
}));
//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/browser.js
var require_browser = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* This is the web browser implementation of `debug()`.
	*/
	exports.formatArgs = formatArgs;
	exports.save = save;
	exports.load = load;
	exports.useColors = useColors;
	exports.storage = localstorage();
	exports.destroy = (() => {
		let warned = false;
		return () => {
			if (!warned) {
				warned = true;
				console.warn("Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
			}
		};
	})();
	/**
	* Colors.
	*/
	exports.colors = [
		"#0000CC",
		"#0000FF",
		"#0033CC",
		"#0033FF",
		"#0066CC",
		"#0066FF",
		"#0099CC",
		"#0099FF",
		"#00CC00",
		"#00CC33",
		"#00CC66",
		"#00CC99",
		"#00CCCC",
		"#00CCFF",
		"#3300CC",
		"#3300FF",
		"#3333CC",
		"#3333FF",
		"#3366CC",
		"#3366FF",
		"#3399CC",
		"#3399FF",
		"#33CC00",
		"#33CC33",
		"#33CC66",
		"#33CC99",
		"#33CCCC",
		"#33CCFF",
		"#6600CC",
		"#6600FF",
		"#6633CC",
		"#6633FF",
		"#66CC00",
		"#66CC33",
		"#9900CC",
		"#9900FF",
		"#9933CC",
		"#9933FF",
		"#99CC00",
		"#99CC33",
		"#CC0000",
		"#CC0033",
		"#CC0066",
		"#CC0099",
		"#CC00CC",
		"#CC00FF",
		"#CC3300",
		"#CC3333",
		"#CC3366",
		"#CC3399",
		"#CC33CC",
		"#CC33FF",
		"#CC6600",
		"#CC6633",
		"#CC9900",
		"#CC9933",
		"#CCCC00",
		"#CCCC33",
		"#FF0000",
		"#FF0033",
		"#FF0066",
		"#FF0099",
		"#FF00CC",
		"#FF00FF",
		"#FF3300",
		"#FF3333",
		"#FF3366",
		"#FF3399",
		"#FF33CC",
		"#FF33FF",
		"#FF6600",
		"#FF6633",
		"#FF9900",
		"#FF9933",
		"#FFCC00",
		"#FFCC33"
	];
	/**
	* Currently only WebKit-based Web Inspectors, Firefox >= v31,
	* and the Firebug extension (any Firefox version) are known
	* to support "%c" CSS customizations.
	*
	* TODO: add a `localStorage` variable to explicitly enable/disable colors
	*/
	function useColors() {
		if (typeof window !== "undefined" && window.process && (window.process.type === "renderer" || window.process.__nwjs)) return true;
		if (typeof navigator !== "undefined" && navigator.userAgent && navigator.userAgent.toLowerCase().match(/(edge|trident)\/(\d+)/)) return false;
		let m;
		return typeof document !== "undefined" && document.documentElement && document.documentElement.style && document.documentElement.style.WebkitAppearance || typeof window !== "undefined" && window.console && (window.console.firebug || window.console.exception && window.console.table) || typeof navigator !== "undefined" && navigator.userAgent && (m = navigator.userAgent.toLowerCase().match(/firefox\/(\d+)/)) && parseInt(m[1], 10) >= 31 || typeof navigator !== "undefined" && navigator.userAgent && navigator.userAgent.toLowerCase().match(/applewebkit\/(\d+)/);
	}
	/**
	* Colorize log arguments if enabled.
	*
	* @api public
	*/
	function formatArgs(args) {
		args[0] = (this.useColors ? "%c" : "") + this.namespace + (this.useColors ? " %c" : " ") + args[0] + (this.useColors ? "%c " : " ") + "+" + module.exports.humanize(this.diff);
		if (!this.useColors) return;
		const c = "color: " + this.color;
		args.splice(1, 0, c, "color: inherit");
		let index = 0;
		let lastC = 0;
		args[0].replace(/%[a-zA-Z%]/g, (match) => {
			if (match === "%%") return;
			index++;
			if (match === "%c") lastC = index;
		});
		args.splice(lastC, 0, c);
	}
	/**
	* Invokes `console.debug()` when available.
	* No-op when `console.debug` is not a "function".
	* If `console.debug` is not available, falls back
	* to `console.log`.
	*
	* @api public
	*/
	exports.log = console.debug || console.log || (() => {});
	/**
	* Save `namespaces`.
	*
	* @param {String} namespaces
	* @api private
	*/
	function save(namespaces) {
		try {
			if (namespaces) exports.storage.setItem("debug", namespaces);
			else exports.storage.removeItem("debug");
		} catch (error) {}
	}
	/**
	* Load `namespaces`.
	*
	* @return {String} returns the previously persisted debug modes
	* @api private
	*/
	function load() {
		let r;
		try {
			r = exports.storage.getItem("debug") || exports.storage.getItem("DEBUG");
		} catch (error) {}
		if (!r && typeof process !== "undefined" && "env" in process) r = process.env.DEBUG;
		return r;
	}
	/**
	* Localstorage attempts to return the localstorage.
	*
	* This is necessary because safari throws
	* when a user disables cookies/localstorage
	* and you attempt to access it.
	*
	* @return {LocalStorage}
	* @api private
	*/
	function localstorage() {
		try {
			return localStorage;
		} catch (error) {}
	}
	module.exports = require_common()(exports);
	const { formatters } = module.exports;
	/**
	* Map %j to `JSON.stringify()`, since no Web Inspectors do that by default.
	*/
	formatters.j = function(v) {
		try {
			return JSON.stringify(v);
		} catch (error) {
			return "[UnexpectedJSONParseError]: " + error.message;
		}
	};
}));
//#endregion
//#region ../../node_modules/.pnpm/supports-color@10.2.2/node_modules/supports-color/index.js
var supports_color_exports = /* @__PURE__ */ __exportAll({
	createSupportsColor: () => createSupportsColor,
	default: () => supportsColor
});
function hasFlag(flag, argv = globalThis.Deno ? globalThis.Deno.args : process$1.argv) {
	const prefix = flag.startsWith("-") ? "" : flag.length === 1 ? "-" : "--";
	const position = argv.indexOf(prefix + flag);
	const terminatorPosition = argv.indexOf("--");
	return position !== -1 && (terminatorPosition === -1 || position < terminatorPosition);
}
function envForceColor() {
	if (!("FORCE_COLOR" in env)) return;
	if (env.FORCE_COLOR === "true") return 1;
	if (env.FORCE_COLOR === "false") return 0;
	if (env.FORCE_COLOR.length === 0) return 1;
	const level = Math.min(Number.parseInt(env.FORCE_COLOR, 10), 3);
	if (![
		0,
		1,
		2,
		3
	].includes(level)) return;
	return level;
}
function translateLevel(level) {
	if (level === 0) return false;
	return {
		level,
		hasBasic: true,
		has256: level >= 2,
		has16m: level >= 3
	};
}
function _supportsColor(haveStream, { streamIsTTY, sniffFlags = true } = {}) {
	const noFlagForceColor = envForceColor();
	if (noFlagForceColor !== void 0) flagForceColor = noFlagForceColor;
	const forceColor = sniffFlags ? flagForceColor : noFlagForceColor;
	if (forceColor === 0) return 0;
	if (sniffFlags) {
		if (hasFlag("color=16m") || hasFlag("color=full") || hasFlag("color=truecolor")) return 3;
		if (hasFlag("color=256")) return 2;
	}
	if ("TF_BUILD" in env && "AGENT_NAME" in env) return 1;
	if (haveStream && !streamIsTTY && forceColor === void 0) return 0;
	const min = forceColor || 0;
	if (env.TERM === "dumb") return min;
	if (process$1.platform === "win32") {
		const osRelease = os.release().split(".");
		if (Number(osRelease[0]) >= 10 && Number(osRelease[2]) >= 10586) return Number(osRelease[2]) >= 14931 ? 3 : 2;
		return 1;
	}
	if ("CI" in env) {
		if ([
			"GITHUB_ACTIONS",
			"GITEA_ACTIONS",
			"CIRCLECI"
		].some((key) => key in env)) return 3;
		if ([
			"TRAVIS",
			"APPVEYOR",
			"GITLAB_CI",
			"BUILDKITE",
			"DRONE"
		].some((sign) => sign in env) || env.CI_NAME === "codeship") return 1;
		return min;
	}
	if ("TEAMCITY_VERSION" in env) return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/.test(env.TEAMCITY_VERSION) ? 1 : 0;
	if (env.COLORTERM === "truecolor") return 3;
	if (env.TERM === "xterm-kitty") return 3;
	if (env.TERM === "xterm-ghostty") return 3;
	if (env.TERM === "wezterm") return 3;
	if ("TERM_PROGRAM" in env) {
		const version = Number.parseInt((env.TERM_PROGRAM_VERSION || "").split(".")[0], 10);
		switch (env.TERM_PROGRAM) {
			case "iTerm.app": return version >= 3 ? 3 : 2;
			case "Apple_Terminal": return 2;
		}
	}
	if (/-256(color)?$/i.test(env.TERM)) return 2;
	if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i.test(env.TERM)) return 1;
	if ("COLORTERM" in env) return 1;
	return min;
}
function createSupportsColor(stream, options = {}) {
	return translateLevel(_supportsColor(stream, {
		streamIsTTY: stream && stream.isTTY,
		...options
	}));
}
var env, flagForceColor, supportsColor;
var init_supports_color = __esmMin((() => {
	({env} = process$1);
	if (hasFlag("no-color") || hasFlag("no-colors") || hasFlag("color=false") || hasFlag("color=never")) flagForceColor = 0;
	else if (hasFlag("color") || hasFlag("colors") || hasFlag("color=true") || hasFlag("color=always")) flagForceColor = 1;
	supportsColor = {
		stdout: createSupportsColor({ isTTY: tty.isatty(1) }),
		stderr: createSupportsColor({ isTTY: tty.isatty(2) })
	};
}));
//#endregion
//#region ../../node_modules/.pnpm/debug@4.4.3/node_modules/debug/src/node.js
var require_node = /* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* Module dependencies.
	*/
	const tty$1 = __require("tty");
	const util = __require("util");
	/**
	* This is the Node.js implementation of `debug()`.
	*/
	exports.init = init;
	exports.log = log;
	exports.formatArgs = formatArgs;
	exports.save = save;
	exports.load = load;
	exports.useColors = useColors;
	exports.destroy = util.deprecate(() => {}, "Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
	/**
	* Colors.
	*/
	exports.colors = [
		6,
		2,
		3,
		4,
		5,
		1
	];
	try {
		const supportsColor = (init_supports_color(), __toCommonJS(supports_color_exports));
		if (supportsColor && (supportsColor.stderr || supportsColor).level >= 2) exports.colors = [
			20,
			21,
			26,
			27,
			32,
			33,
			38,
			39,
			40,
			41,
			42,
			43,
			44,
			45,
			56,
			57,
			62,
			63,
			68,
			69,
			74,
			75,
			76,
			77,
			78,
			79,
			80,
			81,
			92,
			93,
			98,
			99,
			112,
			113,
			128,
			129,
			134,
			135,
			148,
			149,
			160,
			161,
			162,
			163,
			164,
			165,
			166,
			167,
			168,
			169,
			170,
			171,
			172,
			173,
			178,
			179,
			184,
			185,
			196,
			197,
			198,
			199,
			200,
			201,
			202,
			203,
			204,
			205,
			206,
			207,
			208,
			209,
			214,
			215,
			220,
			221
		];
	} catch (error) {}
	/**
	* Build up the default `inspectOpts` object from the environment variables.
	*
	*   $ DEBUG_COLORS=no DEBUG_DEPTH=10 DEBUG_SHOW_HIDDEN=enabled node script.js
	*/
	exports.inspectOpts = Object.keys(process.env).filter((key) => {
		return /^debug_/i.test(key);
	}).reduce((obj, key) => {
		const prop = key.substring(6).toLowerCase().replace(/_([a-z])/g, (_, k) => {
			return k.toUpperCase();
		});
		let val = process.env[key];
		if (/^(yes|on|true|enabled)$/i.test(val)) val = true;
		else if (/^(no|off|false|disabled)$/i.test(val)) val = false;
		else if (val === "null") val = null;
		else val = Number(val);
		obj[prop] = val;
		return obj;
	}, {});
	/**
	* Is stdout a TTY? Colored output is enabled when `true`.
	*/
	function useColors() {
		return "colors" in exports.inspectOpts ? Boolean(exports.inspectOpts.colors) : tty$1.isatty(process.stderr.fd);
	}
	/**
	* Adds ANSI color escape codes if enabled.
	*
	* @api public
	*/
	function formatArgs(args) {
		const { namespace: name, useColors } = this;
		if (useColors) {
			const c = this.color;
			const colorCode = "\x1B[3" + (c < 8 ? c : "8;5;" + c);
			const prefix = `  ${colorCode};1m${name} \u001B[0m`;
			args[0] = prefix + args[0].split("\n").join("\n" + prefix);
			args.push(colorCode + "m+" + module.exports.humanize(this.diff) + "\x1B[0m");
		} else args[0] = getDate() + name + " " + args[0];
	}
	function getDate() {
		if (exports.inspectOpts.hideDate) return "";
		return (/* @__PURE__ */ new Date()).toISOString() + " ";
	}
	/**
	* Invokes `util.formatWithOptions()` with the specified arguments and writes to stderr.
	*/
	function log(...args) {
		return process.stderr.write(util.formatWithOptions(exports.inspectOpts, ...args) + "\n");
	}
	/**
	* Save `namespaces`.
	*
	* @param {String} namespaces
	* @api private
	*/
	function save(namespaces) {
		if (namespaces) process.env.DEBUG = namespaces;
		else delete process.env.DEBUG;
	}
	/**
	* Load `namespaces`.
	*
	* @return {String} returns the previously persisted debug modes
	* @api private
	*/
	function load() {
		return process.env.DEBUG;
	}
	/**
	* Init logic for `debug` instances.
	*
	* Create a new `inspectOpts` object in case `useColors` is set
	* differently for a particular `debug` instance.
	*/
	function init(debug) {
		debug.inspectOpts = {};
		const keys = Object.keys(exports.inspectOpts);
		for (let i = 0; i < keys.length; i++) debug.inspectOpts[keys[i]] = exports.inspectOpts[keys[i]];
	}
	module.exports = require_common()(exports);
	const { formatters } = module.exports;
	/**
	* Map %o to `util.inspect()`, all on a single line.
	*/
	formatters.o = function(v) {
		this.inspectOpts.colors = this.useColors;
		return util.inspect(v, this.inspectOpts).split("\n").map((str) => str.trim()).join(" ");
	};
	/**
	* Map %O to `util.inspect()`, allowing multiple lines if needed.
	*/
	formatters.O = function(v) {
		this.inspectOpts.colors = this.useColors;
		return util.inspect(v, this.inspectOpts);
	};
}));
//#endregion
//#region src/utils/logger.ts
var import_src = /* @__PURE__ */ __toESM((/* @__PURE__ */ __commonJSMin(((exports, module) => {
	/**
	* Detect Electron renderer / nwjs process, which is node, but we should
	* treat as a browser.
	*/
	if (typeof process === "undefined" || process.type === "renderer" || process.browser === true || process.__nwjs) module.exports = require_browser();
	else module.exports = require_node();
})))(), 1);
const logger = O;
const debug = (0, import_src.default)("nuxi");
//#endregion
export { q as _, Et as a, _t as c, gt as d, mt as f, wt as g, st as h, supportsColor as i, at as l, pt as m, logger as n, Gt as o, ot as p, init_supports_color as r, Ot as s, debug as t, fe as u };
